src/nvim/api/vim.c

```diff
@@ -164,6 +164,54 @@ Object vim_eval(String str, Error *err)
   return rv;
 }
 
+/// Call the given function with the given arguments stored in an array.
+///
+/// @param functionName The function to call
+/// @param args the functions arguments packed in an Array
+/// @param[out] err Details of an error that may have occurred
+/// @return The result of the function call if it was successful
+Object vim_call_function(String functionName, Array args, Error *err)
+  FUNC_ATTR_DEFERRED
+{
+  Object retv = OBJECT_INIT;
+  if (args.size > MAX_FUNC_ARGS) {
+    api_set_error(err, Validation,
+      _("Function called with too many arguments."));
+    return retv;
+  }
+
+  // Convert the arguments in args from Object to typval_T values
+  typval_T vim_args[MAX_FUNC_ARGS + 1];
+  size_t i = 0;  // also used for freeing the variables
+  for (; i < args.size; i++) {
+    if (!object_to_vim(args.items[i], &vim_args[i], err)) {
+      goto free_vim_args;
+    }
+  }
+
+  try_start();
+  // Call the function
+  typval_T tmp;
+  int dummy;
+  int r = call_func((char_u*) functionName.data, (int)functionName.size,
+            &tmp, (int)args.size, vim_args,
```

Comment:
Style guide says that you either need to align next argument after opening parenthesis that surrounds it or put all arguments on the lines after function name: http://neovim.io/develop/style-guide.xml#Function_Calls (with a soft suggestion to use “one argument per line” variant; I personally in this case prefer either arguments grouped by meaning or simply “wrap at 80” without any grouping).

I.e. this and next lines are too dedented.